contracts/Marketplace.sol

```diff
@@ -359,176 +338,96 @@ contract Marketplace is SlotReservations, Proofs, StateRetrieval, Endian {
     }
   }
 
-  /**
-     * @notice Abandons the slot without returning collateral, effectively slashing the
-     entire collateral.
-   * @param slotId SlotId of the slot to free.
-   * @dev _slots[slotId] is deleted, resetting _slots[slotId].currentCollateral
-     to 0.
-  */
+  /// Abandons the slot, burns all associated tokens
   function _forciblyFreeSlot(SlotId slotId) internal {
     Slot storage slot = _slots[slotId];
     RequestId requestId = slot.requestId;
     RequestContext storage context = _requestContexts[requestId];
 
-    // We need to refund the amount of payout of the current node to the `fundsToReturnToClient` so
-    // we keep correctly the track of the funds that needs to be returned at the end.
-    context.fundsToReturnToClient += _slotPayout(requestId, slot.filledAt);
+    Request storage request = _requests[requestId];
+
+    TokensPerSecond rate = request.ask.pricePerSlotPerSecond();
+    uint128 collateral = request.collateralPerSlot();
+    uint128 repairReward = _config.collateral.repairReward(collateral);
+
+    FundId fund = requestId.asFundId();
+    AccountId hostAccount = _vault.hostAccount(slot.host, slot.slotIndex);
+    AccountId clientAccount = _vault.clientAccount(request.client);
+    _vault.flow(fund, hostAccount, clientAccount, rate);
+    _vault.transfer(fund, hostAccount, clientAccount, repairReward);
+    _vault.burnAccount(fund, hostAccount);
 
     _removeFromMySlots(slot.host, slotId);
     _reservations[slotId].clear(); // We purge all the reservations for the slot
     slot.state = SlotState.Repair;
-    slot.filledAt = 0;
     slot.currentCollateral = 0;
     slot.host = address(0);
     context.slotsFilled -= 1;
     emit SlotFreed(requestId, slot.slotIndex);
     _resetMissingProofs(slotId);
 
-    Request storage request = _requests[requestId];
     uint256 slotsLost = request.ask.slots - context.slotsFilled;
     if (
       slotsLost > request.ask.maxSlotLoss &&
       context.state == RequestState.Started
     ) {
       context.state = RequestState.Failed;
-      context.endsAt = uint64(block.timestamp) - 1;
+      _vault.freezeFund(fund);
+
       emit RequestFailed(requestId);
     }
   }
 
   function _payoutSlot(
     RequestId requestId,
-    SlotId slotId,
-    address rewardRecipient,
-    address collateralRecipient
+    SlotId slotId
   ) private requestIsKnown(requestId) {
     RequestContext storage context = _requestContexts[requestId];
     Request storage request = _requests[requestId];
     context.state = RequestState.Finished;
     Slot storage slot = _slots[slotId];
+    slot.currentCollateral = 0;
```

Comment:
Why do you do that? The Request is Finished anyway already no?

Actually this is really used only for tracking purpose for the host to know how much collateral does he have left right?